tests/containers/upload_container_images_to_ecr.pm

```diff
@@ -23,11 +23,18 @@ sub run {
 
     my ($untested_images, $released_images) = get_suse_container_urls();
 
-    my $provider = $self->provider_factory();
-    my $image    = $untested_images->[0];
-    my $tag      = join('-', get_var('PUBLIC_CLOUD_RESOURCE_NAME'), get_current_job_id());
+    my $provider      = $self->provider_factory();
+    $self->{provider} = $provider;
```

Comment:
isn't this already done by `provider_factory()`?

Reply:
This `$self` is used when you need to preserve the variable into other tests.

Reply:
@pdostal not sure, normally we use the args from the `run()` method to move the provider from one to the other test.
I think `$self->{provider}` is needed to do a proper cleanup in `cleanup()` function.

Reply:
Ah, true. My statement would be valit for `$args` not for `$self`.